railties/test/generators/plugin_generator_test.rb

```diff
@@ -92,6 +92,28 @@ def test_initializes_git_repo
     assert_directory ".git"
   end
 
+  def test_initializes_git_repo_with_main_branch
+    run_generator
+    assert_file ".git/HEAD", /main/
+  end
+
+  def test_version_control_initializes_git_repo_with_user_default_branch
+    git_version = `git --version`[/\d+.\d+.\d+/]
+    return if Gem::Version.new(git_version) < Gem::Version.new("2.28.0")
+
+    current_default_branch = `git config --global init.defaultBranch`
+    `git config --global init.defaultBranch master`
+
+    run_generator
+    assert_file ".git/HEAD", /master/
+
+    if current_default_branch.strip.empty?
+      `git config --global --unset init.defaultBranch`
+    else
+      `git config --global init.defaultBranch #{current_default_branch}`
+    end
```

Comment:
same here. anytime you modify a config that would permanently affect the machine running the test suite, setting the config back to original should be ensured (in an ensure block)